ocaml/xapi/xapi_session.ml

```diff
@@ -791,11 +791,18 @@ module Caching = struct
      disabled. This function exists to delay the construction of the
      cache, as Xapi_globs configuration is not guaranteed to have been
      populated before the top-level code of this module is executed. *)
-  let get_or_init_cache () =
-    if not !Xapi_globs.external_authentication_cache_enabled then
+  let get_or_init_cache ~__context =
+    (* read db here, Helpers.get_pool *)
+    let pool = Helpers.get_pool ~__context in
+    let cache_enabled =
+      Db.Pool.get_ext_auth_cache_enabled ~__context ~self:pool
+    in
+    if not cache_enabled then
       None
     else
-      let capacity = !Xapi_globs.external_authentication_cache_size in
+      let capacity =
+        Db.Pool.get_ext_auth_cache_size ~__context ~self:pool |> Int64.to_int
```

Comment:
What is the expected outcome when the values of the new DB fields are changed?

I looks like a change in the field is ignored once the cache has been initialised, so that it would only take effect after restarting xapi on the coordinator?

Also, if you change from enabled to disabled, the cache isn't cleared?

Reply:
> So, to some extent, these don't have an immediate effect on the the behaviour of the cache (if present).

This may be fine. But could some logs be added so that we could know if the cache is taking effect or not. E.g. from a bug report, it would be helpful to understand the actual situation.